sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -148,6 +150,81 @@
+            IntelliJAuthMethodDetails authDetails = cacheAccessor.getAuthDetailsIfAvailable();
+            String authType = authDetails.getAuthMethod();
+            if (authType.equalsIgnoreCase("SP")) {
+                Map<String, String> spDetails = cacheAccessor
+                    .getIntellijServicePrincipalDetails(authDetails.getCredFilePath());
+                String authorityUrl = spDetails.get("authURL") + spDetails.get("tenant");
+                try {
+                    ConfidentialClientApplication.Builder applicationBuilder =
+                        ConfidentialClientApplication.builder(spDetails.get("client"),
+                            ClientCredentialFactory.createFromSecret(spDetails.get("key")))
+                            .authority(authorityUrl);
+
+                    // If http pipeline is available, then it should override the proxy options if any configured.
+                    if (httpPipelineAdapter != null) {
+                        applicationBuilder.httpClient(httpPipelineAdapter);
+                    } else if (options.getProxyOptions() != null) {
+                        applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
+                    }
+
+                    if (options.getExecutorService() != null) {
+                        applicationBuilder.executorService(options.getExecutorService());
+                    }
+
+                    ConfidentialClientApplication application = applicationBuilder.build();
+                    return Mono.fromFuture(application.acquireToken(
+                        ClientCredentialParameters.builder(new HashSet<>(request.getScopes()))
+                            .build()))
+                               .map(ar -> new MsalToken(ar, options));
+                } catch (MalformedURLException e) {
+                    return Mono.error(e);
+                }
+            } else if (authType.equalsIgnoreCase("DC")) {
+                String authHost = cacheAccessor.getAzureAuthHost(authDetails.getAzureEnv())
+                                      .replaceAll("/+$", "") + "/organizations/" + tenantId;
+
+                PublicClientApplication.Builder applicationBuilder = PublicClientApplication.builder(clientId)
+                    .authority(authHost);
+
+                if (httpPipelineAdapter != null) {
+                    applicationBuilder.httpClient(httpPipelineAdapter);
+                } else if (options.getProxyOptions() != null) {
+                    applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
+                }
+
+                if (options.getExecutorService() != null) {
+                    applicationBuilder.executorService(options.getExecutorService());
+                }
+
+                PublicClientApplication publicClientApplication = applicationBuilder
+                                                                      .build();
```

Comment:
Same comment here as on the VsCodeCredential. You're creating a new PublicClientApplication (and a new ConfidentialClientApplication above) each time you try to get a token so there will be no caching. We don't necessarily have to fix this before merging it for our preview release but if not we should create an issue to track the work.